build.install4j

```diff
@@ -535,6 +699,70 @@
                 </serializedBean>
                 <condition />
               </action>
+              <action name="" id="176" customizedId="" beanClass="com.install4j.runtime.beans.actions.registry.DeleteRegistryItemAction" enabled="true" commentSet="false" comment="" actionElevationType="elevated" rollbackBarrier="false" multiExec="false" failureStrategy="1" errorMessage="Failed to unregister the protocol handler">
+                <serializedBean>
+                  <java class="java.beans.XMLDecoder">
+                    <object class="com.install4j.runtime.beans.actions.registry.DeleteRegistryItemAction">
+                      <void property="keyName">
+                        <string>Software\Classes\triplea</string>
+                      </void>
+                      <void property="onlyIfEmpty">
+                        <boolean>false</boolean>
+                      </void>
+                      <void property="registryRoot">
+                        <object class="java.lang.Enum" method="valueOf">
+                          <class>com.install4j.api.windows.RegistryRoot</class>
+                          <string>HKEY_CURRENT_USER</string>
+                        </object>
+                      </void>
+                    </object>
+                  </java>
+                </serializedBean>
+                <condition>Util.isWindows() &amp;&amp; ((String)WinRegistry.getValue(RegistryRoot.HKEY_CURRENT_USER, "Software\\Classes\\triplea\\shell\\open\\command", "")).equalsIgnoreCase("\"" + context.getVariable("sys.installationDir") + "\\TripleA.exe\" \"%1\"")</condition>
```

Comment:
Just curious about two things...

1. Why was it necessary to add `Util.isWindows()` to the condition here but not in the `SetRegistryValueAction`s?  I would have assumed as with the `SetRegistryValueAction`s, install4j "knows" that `DeleteRegistryItemAction` can only run on Windows.  Is it because it is a compound condition and install4j will evaluate the second sub-expression regardless of the OS (and thus fail on non-Windows systems)?
1. Is the second sub-expression (`WinRegistry.getValue(...`) the magic that leaves a new installation's protocol handler intact when you uninstall an older version as you described [here](https://github.com/triplea-game/triplea/pull/1783#issuecomment-307525201)?